src/web/src/getlog_GET.ecpp

```diff
@@ -279,238 +280,392 @@ class _clonedcode_Compressor
         void clear()
         {
           _zbody.makeEmpty();
-          /* This is in newer codebase : _deflator.reinitialize(); which does "deflateReset(&_stream)" */
+          // This is in newer codebase : _deflator.reinitialize(); which does "deflateReset(&_stream)"
           _crc = 0;
           _size = 0;
         }
 }; // class _clonedcode_Compressor
 
+// journald logs export to logfile
+int exec_journalctl_command (const std::string& logfile)
+{
+    // 'www-data' user must be owned by 'systemd-journal' group,
+    // to let journalctl access journald files (see fty-core)
+    // usermod -a -G systemd-journal www-data
+    // avoid wrapper usage /usr/libexec/bios/journalctl
+    std::string cmd;
+    cmd = "/bin/journalctl -D /var/log/journal --no-pager";
+    cmd.append(" -o cat"); // output mode
+    cmd.append(" --since \"3 hours ago\" --until \"now\"");
+    cmd.append(" > ").append(logfile);
+
+    log_debug("running journalctl command to generate %s", logfile.c_str());
+    int r = system(cmd.c_str());
+    if (r != 0) log_error("journalctl command returns %d ", r);
+    else log_debug("journalctl command returns %d ", r);
+
+    return r;
+}
 
 </%pre>
 <%request scope="global">
 UserInfo user;
 </%request>
 <%cpp>
+
     // permission check
     static const std::map <BiosProfile, std::string> PERMISSIONS = {
-            {BiosProfile::Admin,     "R"}
-            };
+        {BiosProfile::Admin, "R"}
+    };
     CHECK_USER_PERMISSIONS_OR_DIE (PERMISSIONS);
-    /* Note: the http_die() family does its own
-     *  reply.setContentType("application/json;charset=UTF-8");
-     * rituals if needed, so undecided callers like this one do not */
-
-    /* NOTE: At a later stage we might do other logs than pre-existing files -
-     * e.g. exports from journalctl, so logNAME and logFILE are logically and
-     * maybe functionally different beasts below... */
-    std::string logfile; /* Full path as set for supported logfiles below */
-    std::string list_lognames = "no";
-    std::string logname_ext;
-    std::string logname_base;
+
+    // Note: the http_die() family does its own
+    //  reply.setContentType("application/json;charset=UTF-8");
+    // rituals if needed, so undecided callers like this one do not
+
+    // NOTE: At a later stage we might do other logs than pre-existing files -
+    // e.g. exports from journalctl, so logNAME and logFILE are logically and
+    // maybe functionally different beasts below...
+
+    // supported (baseName, filePath) pairs
+    struct {
+        std::string baseName;
+        std::string filePath;
+    } logsData [] = {
+        { "messages",       "/var/log/messages" },
+        { "www-audit",      "/var/log/tntnet/www-audit.log" },
+        { "emc4j-audit",    "/var/log/etn-emc4j-ipm2/karaf.log" },
+        { "journald",       "/tmp/journald.log" }, // gen. by journalctl
+        { "ALL",            "/tmp/getlog_all.tar" }, // archive,all logfiles above
```

Comment:
any reason to use upper case for "ALL" and not the others?

Reply:
not really, just a manner to strongly differentiate the 'all logs' feature.